### PR TITLE
Install / configured material-ui for React

### PR DESCRIPTION
**Pull request by Tybot204**
This is dependent on https://github.com/Cathartes/records-spa/pull/1.

Material UI is a great way to easily implement common elements of websites. More info can be found here: http://www.material-ui.com. It's Angular Material, but for React.

I intentionally have not been implementing any real styling here, and aimed to only get initial configuration and a quick POC done so that anyone else can start making the site look nicer once there are a few basic components in.

### CODE REVIEW

**Comment by NicksPatties** [comment]
If we're not sure what the actual design of the pages are going to look like, then how can we kow for sure that material-react is the way to go?

Also, warning about using this package while I was playing with it on the developer's site: it appears that the styles for each if the components are injected into the style tags of whatever html is being rendered, which means that we cannot define styles in css classes, and override the styles for components in case we need to.

**Comment by Tybot204** [comment]
@NicksPatties This is more about ease of use and speed of development for now. My thoughts were use a library that makes our site look nice for now, take an in depth style pass later. I am personally a fan of Material Design principals, and @KTSCode has also expressed his love of this library to me.

I've used Angular Material for projects in the past so it means slightly less learning for me to do too ;)

In all seriousness though, @jlaforge and myself do not care what the site looks like for the time being. We will create mock ups when possible, but since we are the only 2 that will be using this site initially, we just need a functional, decent looking set of pages.

**Comment by NicksPatties** [comment]
Ah, it looks like you can pass classes into these components with the `cssClass` prop. But hold on...
> Note that CSS properties defined inline are given priority over those defined in a CSS class. You need to use !important to take precedence over the inline style.

So either using inline styles, or classes with `!important` all over their style declarations. Both options sound like a headache to deal with.
* The inline styles route
  * Need to either remember to copy-paste the styles in components that we want (or more ideally, import a JS object into the component file to pass those styles into components), but now we have styles in both `.css` and `.js` files, which will be annoying to track down and make changes.
* The classes route
  * Need to *actually* copy paste `!important` over and over again.

I understand that putting this package in here is useful for just getting a POC out, but I'm just thinking of all the additional work that would need to be done in the future once we try to add cool looking stuff.

**Comment by Tybot204** [comment]
@NicksPatties check out the 1.0 beta release. Looks a bit more friendly to us. I have it installed here since I ran into issues already with the pre-1.0 version.

https://material-ui-1dab0.firebaseapp.com/

File: tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "forceConsistentCasingInFileNames": true,
     "noImplicitReturns": true,
     "noImplicitThis": true,
-    "noImplicitAny": true,
+    "noImplicitAny": false,
```
Comment (Tybot204) [inline tsconfig.json]: This is unfortunately a requirement for the time being until Material UI is released in full. If we don't use Material UI and we have the man power to style pages from scratch, I will turn this back on.

**Pull request merged**